CppSamples/Maps/CreateDynamicBasemapGallery/BasemapStyleListModel.cpp

```diff
@@ -60,7 +61,11 @@ void BasemapStyleListModel::insertItemsIntoGallery(const QList<BasemapStyleInfo*
     m_previews.clear();
     for (const BasemapStyleInfo* info : infos)
     {
-        m_previews.append(std::move(info));
+        connect(info, &BasemapStyleInfo::thumbnailUrlChanged, this, [this, info]() {
```

Comment:
Is `info` lifetime is long enough to be used in connection? Maybe moving the index outside the lambda function is safer?

```cpp
const auto index = m_previews.indexOf(info);
connect(info, &BasemapStyleInfo::thumbnailUrlChanged, this, [this, index]() {
...
```

Reply:
I like @GuillaumeBelz's idea, but I would augment that to add that we should just switch to a traditional for loop with index instead of repeatedly calling `indexOf` which is not a cheap call.

Reply:
Used traditional for loop